cpp/src/gandiva/arrow.h

```diff
@@ -48,6 +51,14 @@ using ArrayDataVector = std::vector<ArrayDataPtr>;
 using Status = arrow::Status;
 using StatusCode = arrow::StatusCode;
 
+static inline bool is_decimal_128(DataTypePtr type) {
+  if (type->id() == arrow::Type::DECIMAL) {
+    auto decimal_type = dynamic_cast<arrow::DecimalType*>(type.get());
```

Comment:
You can use `arrow::internal::checked_cast` (see `util/checked_cast.h`).

Reply:
will fix.